node/node.go

```diff
@@ -78,7 +78,8 @@ type Node struct {
 	IndexerService *txindex.IndexerService
 
 	// shared context for all dymint components
-	ctx context.Context
+	ctx    context.Context
+	cancel context.CancelFunc
```

Comment:
The node object is the source of the ctx, and it passes it to it's components by arg
node.ctx is the only place where it lives in an object, so it could be cancelable

Reply:
what are we coordinating with the `ctx` here, apart from stopping prometheus?

Reply:
P2P, blockManager,